shell/common/pipeline.h

```diff
@@ -185,9 +192,13 @@ class Pipeline : public fml::RefCountedThreadSafe<Pipeline<R>> {
   void ProducerCommitFront(ResourcePtr resource, size_t trace_id) {
     {
       std::scoped_lock lock(queue_mutex_);
-      queue_.emplace_front(std::move(resource), trace_id);
-      while (queue_.size() > depth_) {
-        queue_.pop_back();
+      if (queue_.size() > 0) {
+        FML_LOG(INFO) << "ignoring CommitFront in favor of existing queue entry";
```

Comment:
Is there any consequence or downside of throwing away the resource when the queue is nonempty?

Reply:
This is in the `ProduceToFront` (of queue) call. It is called when we go to rasterize a LayerTree and are told to resubmit it instead. When we had a queue, that LayerTree came off the front of the queue and so that's where it should go back on. In the new system I'm assuming that something in the queue will be newer than what we had just been processing and so I just let it fall on the floor.

These two methods to add something to the queue could all be replaced by a single version of "ProduceCommit" that just compares the target times of the incoming LayerTree with the target time of the one that is already waiting and keeps only the newest version.

Reply:
I'll add that the only source of "kResubmit" in the code I could find came from the iOS platform view code and looked to be related to a change in threading resulting from a change in the structure of a frame (due to a change in the embedding?)...

```shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm:      return PostPrerollResult::kResubmitFrame;```